tests/api_test.py

```diff
@@ -1910,14 +1910,18 @@ def unflatten(unused_aux_data, children):
     jax.device_put(CustomNode([0.1]))
 
   def test_vmap_inconsistent_sizes_constructs_proper_error_message(self):
```

Comment:
Rather than replacing the previous test case, I'd suggest leave the current test as-is and add a new test method with the problematic case you reported in #24406 (with a comment that it's a regression test for that issue).

Reply:
Done.